src/App.js

```diff
@@ -3,22 +3,25 @@ import TodoList from './TodoList.js';
 import AddToDoForm from './AddTodoForm.js';
 
 
-const useSemiPersistentState = () => {
-
-  const [todoList, setTodoList] = React.useState(JSON.parse(localStorage.getItem("savedTodoList")));
-
-  React.useEffect(() => {localStorage.setItem("savedTodoList", JSON.stringify(todoList));},[todoList]);
-
-  return[todoList, setTodoList];
-
-};
-
-
 
 
 function App() {
-  
-  const [todoList, setTodoList] = useSemiPersistentState();
+    
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsloading] = React.useState(true);
+
+  React.useEffect(() => new Promise((resolve,reject) => setTimeout( () => 
+  resolve({ data: {todoList: JSON.parse(localStorage.getItem("savedTodoList"))}}), 2000))
+  .then(function(result) {
+    console.log(result);
+    setTodoList(result.data.todoList);//is this correct?
+    setIsloading(false);//is this correct?
+  }),[]);
+
+  React.useEffect(() => {
+    if(!isLoading){//is this correct?
```

Comment:
You'll need to update the dependencies of the `useEffect` to include `isLoading`.  So it will be: `[todoList, isLoading]`